filesystem/File.php

```diff
@@ -238,7 +238,7 @@ public function exists() {
 	 */
 	public static function find($filename) {
 		// Get the base file if $filename points to a resampled file
-		$filename = preg_replace('/_resampled\/[^-]+-/', '', $filename);
+		$filename = preg_replace('/_resampled\/(.+\/|[^-]+-)/', '', $filename);
```

Comment:
Before I push my changes, thought I'd ask you opinion on the name of that static method. I started out called it `Image::get_original_filename`, but that ends up not making sense because it makes it look like you can only provide an Image filename as a parameter, while its usages accept any kind of of Filename (pdf, xls,..), and they just want to make sure it doesn't have some _resampled stuff at the from. Feels wrong to put an xls filename into an `Image::get_original_filename` method, so I ended up calling it `Image::strip_resampled_prefix` instead. Make sense?..